pkg/apis/unikorn/v1alpha1/types.go

```diff
@@ -741,3 +741,66 @@ type OpenstackServerSpec struct {
 
 type OpenstackServerStatus struct {
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type FileStorageList struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Items             []FileStorage `json:"items"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope=Namespaced,categories=unikorn
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="protocol",type="string",JSONPath=".spec.protocol"
+// +kubebuilder:printcolumn:name="attachments",type="string",JSONPath=".spec.attachments"
+// +kubebuilder:printcolumn:name="size",type="string",JSONPath=".spec.size"
+// +kubebuilder:printcolumn:name="storageClassID",type="string",JSONPath=".spec.storageClassID"
+// +kubebuilder:printcolumn:name="age",type="date",JSONPath=".metadata.creationTimestamp"
+type FileStorage struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              FileStorageSpec   `json:"spec"`
+	Status            FileStorageStatus `json:"status"`
+}
+
+type FileStorageSpec struct {
+	RegionID string `json:"regionID"`
+	// Name of the FileStorageClass
+	StorageClassID string                `json:"storageClassID"`
+	Size           resource.Quantity     `json:"size"`
+	Attachments    []Attachment          `json:"attachments,omitempty"`
+	Protocol       Protocol              `json:"protocol"`
+	Tags           unikornv1core.TagList `json:"tags,omitempty"`
+
+	// Pause, if true, will inhibit reconciliation.
+	Pause bool `json:"pause,omitempty"`
+	NFS   *NFS `json:"nfs,omitempty"`
+}
+
+type FileStorageStatus struct {
+	Conditions []unikornv1core.Condition `json:"conditions,omitempty"`
+}
+type Attachment struct {
+	NetworkID string `json:"networkID"`
+}
+
+type NFS struct {
+	RootSquash bool `json:"rootSquash,omitempty"`
+}
+
+type FileStorageClass struct {
```

Comment:
Ah yeah, this needs to be a CRD too, with it's List type etc. and all the +kubebuilder nonsense.  We will expect storage classes to be provisioned by the admin via Helm.